### PR TITLE
FS: Profile Context

### PR DESCRIPTION
**Pull request by dproc96**
**What it does**
- Adds a context for storing the user's profile distinct from the user
- Grabs the profile's avatar for the Navbar
- Adds "/profile/me" route to get the logged in user's profile
- Makes a few features we're working on simpler to implement

**What to avoid**
- Cannot test feature until Issue #16 can be merged in

### CODE REVIEW

File: client/src/components/Navbar.js [hunk 1]
```diff
@@ -105,6 +116,8 @@ function getNavigation({ width, classes, anchorEl, handleClick, handleClose }) {
 function Navbar(props) {
   const { user } = useUserContext();
   const classes = useStyles();
+  const { profile } = useProfileContext();
+  console.log(profile)
```
Comment (dahrk) [inline client/src/components/Navbar.js]: extra logging here

Comment (dproc96) [reply client/src/components/Navbar.js]: Oops, missed this one when I was removed console logs, just pushed a new commit

File: client/src/App.js [hunk 2]
```diff
@@ -12,22 +11,25 @@ import ProfileSettings from "./pages/ProfileSettings";
...
-          <PrivateRoute path="/me" component={ProfileSettings} />
-          <PrivateRoute exact path="/become-a-sitter" component={LandingPage} />
-          <PrivateRoute exact path="/my-sitters" component={LandingPage} />
-          <PrivateRoute exact path="/my-jobs" component={LandingPage} />
-          <PrivateRoute exact path="/messages" component={LandingPage} />
+          <ProfileProvider>
+            <Navbar />
+            <Route exact path="/" component={ProfileSettings} />
+            <Route exact path="/signup" component={SignUp} />
+            <Route path="/login" component={Login} />
```
Comment (aeciorc) [inline client/src/App.js]: The auth routes could be outside of the provider, right? Since the user isn't auth yet

Comment (dproc96) [reply client/src/App.js]: I think so, but @Braagaa can you confirm this would still work with your user integration?

Comment (dproc96) [reply client/src/App.js]: Just pushed a change to move those two outside but still in the AuthProvider

Comment (Braagaa) [reply client/src/App.js]: Yep that's fine

**Review by aeciorc — APPROVED** [review]

**Pull request merged**